Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -694,6 +697,21 @@ extension MapboxMap: MapFeatureQueryable {
                                                                     concreteErrorType: MapError.self))
     }
 
+    /// Queries the map for rendered features.
+    ///
+    /// - Parameters:
+    ///   - geometry: The `screen pixel coordinates` (point, line string or box) to query for rendered features.
+    ///   - options: Options for querying rendered features.
+    ///   - completion: Callback called when the query completes
+    func queryRenderedFeatures(for geometry: RenderedQueryGeometry, options: RenderedQueryOptions? = nil, completion: @escaping (Result<[QueriedFeature], Error>) -> Void) {
```

Comment:
Exposing a single QRF method with `RenderedQueryGeometry` will make for quite an inconvenient API for platform developers, e.g:
```swift
mapView.mapboxMap.queryRenderedFeatures(for: RenderedQueryGeometry.fromScreenCoordinate(ScreenCoordinate(x: tapLocation.x, y: tapLocation.y)), completion: { _ in })
```

Let's deprecate existing QRF methods instead adding their cancellable counterparts that use `RenderedQueryGeometry` underneath.
```swift
     func queryRenderedFeatures(at point: CGPoint,
                                options: RenderedQueryOptions?,
                                completion: @escaping (Result<[QueriedFeature], Error>) -> Void) {
        return __map.__queryRenderedFeatures(for: .fromScreenCoordinate(point.screenCoordinate),
                                       options: options ?? RenderedQueryOptions(layerIds: nil, filter: nil),
                                       callback: coreAPIClosureAdapter(for: completion,
                                                                       type: NSArray.self,
                                                                       concreteErrorType: MapError.self)).asCancellable()
}

    func queryRenderedFeatures(for shape: [CGPoint],
                                options: RenderedQueryOptions?,
                                completion: @escaping (Result<[QueriedFeature], Error>) -> Void) -> Cancelable {
    ...
}

     func queryRenderedFeatures(in rect: CGRect,
                                options: RenderedQueryOptions?,
                                completion: @escaping (Result<[QueriedFeature], Error>) -> Void) -> Cancelable {
   ....
}
```